swift/tools/tracing-config.lua

```diff
@@ -3,6 +3,53 @@ function RegisterExtractorPack(id)
     local relativeSwiftExtractor = extractorDirectory .. 'extractor'
     local swiftExtractor = AbsolutifyExtractorPath(id, relativeSwiftExtractor)
 
+    function indexOf(array, value)
+      for i, v in ipairs(array) do
+        if v == value then
+          return i
+        end
+      end
+      return nil
+    end
+
+    -- removes upsupported CLI arg including the following how_many args
+    function strip_unsupported_arg(args, arg, how_many)
+      index = indexOf(args, arg)
+      if index then
+        table.remove(args, index)
+        while (how_many > 0)
+        do
+          table.remove(args, index)
+          how_many = how_many - 1
+        end
+      end
+    end
+
+    function strip_unsupported_args(args)
+      strip_unsupported_arg(args, '-emit-localized-strings', 0)
+      strip_unsupported_arg(args, '-emit-localized-strings-path', 1)
+      strip_unsupported_arg(args, '-stack-check', 0)
+    end
+
+    -- xcodebuild does not always specify the -resource-dir in which case the compiler falls back
+    -- to a resource-dir based on its path
+    -- here we mimic this behavior externally
+    -- without a proper -resource-dir compiler-specific headers cannot be found which leads to
+    -- broken extraction
+    function insert_resource_dir_if_needed(compilerPath, args)
+      resource_dir_index = indexOf(args, '-resource-dir')
+      if resource_dir_index then
+        return
+      end
+      -- derive -resource-dir based on the compilerPath
+      -- e.g.: /usr/bin/swift-frontend -> /usr/bin/../lib/swift
+      local last_slash_index = string.find(compilerPath, "/swift%-frontend")
```

Comment:
We do have the exact filename check before this call, so it might be redundant.

Actually, what I wanted to do here is simply to cut the last path component. What is a better way to do it? I tried reading lua lang docs, but failed at finding the right tool for this 😅

Reply:
maybe then just 
```suggestion
      local last_slash_index = string.find(compilerPath, "/[^/]*$")
```
to find the last `/`?

Reply:
The new regex doesn't look better to me 😅

Reply:
@redsun82 good point! I applied the change!